articles/141_static_remapping.md

```diff
@@ -339,8 +340,8 @@ Within each category, the rules are applied in the order in which the user gave
 **Example of node/namespace remapping order:**
 
 - A node has name `talker`
-- A user gives the rules `talker:__ns:=/my_namespace` then `talker:__node:=foo`
-- The final namespace is the default `/` because the node name remap is applied before the namespace remap
+- A user gives the rules `talker:__ns:=/my_namespace` then `/talker:__node:=foo`
+- The final node name is the default `talker` because the namespace remap is applied before the node name remap
```

Comment:
what if user wants to remap `talker` into namespace `A` and `B`? I believe this is really common use case. I am not so convinced to change order. I think `Node name -> Namespace` would be understandable and usable for user.

Reply:
Thanks for your response. @fujitatomoya 

> what if user wants to remap talker into namespace A and B?

Sorry, I don't catch it. How can we remap one node `talker` into two different namespaces `A` and `B`?

> Node name -> Namespace would be understandable

It has a limitation.
Imageing there are two nodes `publisher_node` and `subscriber_node` in an application.
/publisher_node
/subscriber_node

I want to remap these two nodes as following (https://github.com/ros2/rcl/issues/806),
`/publisher_node` -----> '/ns1/aaa'
`/subscriber_node` -----> '/ns2/aaa'

Reply:
Maybe this case is suited for the real world.
/left_camera        -----> /left/camera
/right_camera      -----> /right/camera

Reply:
> How can we remap one node talker into two different namespaces A and B?

bad example, ignore it... sorry😢

> It seems we can not remap them with current remapping order.
> /left_camera -----> /left/camera
> /right_camera -----> /right/camera

I don't see this is because of remapping order, but implementation. I think we should have some kinda justification to change order to support two nodes with the same name but in different namespaces.

Reply:
> I don't see this is because of remapping order, but implementation.

That's right, users can set two nodes with Node("camera", "/left") and Node("camera", "/right") in implementation.
but we can't ask users to do this, if somebody just wants to use remapping, there is a limitation.

>  some kinda justification 

Actually, the sample about the 'camera' above is the reason why I open this issue.

I found no discussion about why ros2 use this kind of remapping order(node name -> namespace), after checking https://github.com/ros2/rcl/pull/217 and https://github.com/ros2/design/pull/181, it seems that [ros2](https://github.com/ros2/rcl/blob/6f39edf904a6a041099c5deaf313ff975da4c3c8/rcl/src/rcl/node.c#L229-L254) follow the same implementation order to [ros1](https://github.com/ros/ros_comm/blob/84d413fa5689891d871680c18f217a6a16d12731/clients/roscpp/src/libros/this_node.cpp#L122-L134). (If I remember correctly, ros1 doesn't have the remapping order for __name and __ns because it just use one node in an application. )